### PR TITLE
Make Rule KeyColumns configuration mandatory

### PR DESCRIPTION
**Pull request by ryanemerson**
- Make Rule KeyColumns configuration mandatory
- Utilise @PostConstruct to ensure HotRodService is initiated before use

@wburns @jabolina I originally just wanted to create a PR for the first commit, but I kept getting startup failures with the HotRodService hence the additional commit.

### CODE REVIEW

File: src/main/java/io/gingersnapproject/cdc/configuration/Rule.java [hunk 1]
```diff
@@ -15,12 +15,12 @@ public interface Rule {
       @WithDefault("|")
       String plainSeparator();
 
-      Optional<List<String>> keyColumns();
+      List<String> keyColumns();
 
       Optional<List<String>> columns();
```
Comment (jabolina) [inline src/main/java/io/gingersnapproject/cdc/configuration/Rule.java:20]: Should the columns be required?

Comment (ryanemerson) [reply src/main/java/io/gingersnapproject/cdc/configuration/Rule.java:20]: I don't think so. If none are explicitly configured  then we should return all columns in the table.

WDYT to changing the field name to `valueColumns` to make it's purpose more explicit?

Comment (jabolina) [reply src/main/java/io/gingersnapproject/cdc/configuration/Rule.java:20]: +1 for renaming, and also reflects the `keyColumns`.

**Review by jabolina — APPROVED** [review]

**Comment by jabolina** [comment]
Thanks, @ryanemerson

**Pull request merged**